pkg/event/controller.go

```diff
@@ -96,7 +100,21 @@ func (gen *generator) Add(infos ...Info) {
 			logger.V(3).Info("skipping event creation for resource without a name", "kind", info.Kind, "name", info.Name, "namespace", info.Namespace)
 			continue
 		}
-		gen.queue.Add(info)
+		// if the `emit-event` flag is not set
+		// directly add the event to the queue
+		if len(gen.emitEvents) == 0 {
+			gen.queue.Add(info)
+		}
+		// flag is set
+		// only add the event when the reason of event is specified in "emit-events" flag
+		for _, eventReason := range gen.emitEvents {
+			if info.Reason == Reason(eventReason) {
+				logger.V(6).Info("creating event", "kind", info.Kind, "name", info.Name, "namespace", info.Namespace, "reason", info.Reason, "event reason", eventReason)
+				gen.queue.Add(info)
+			} else {
+				logger.V(6).Info("event reason not matched", "kind", info.Kind, "name", info.Name, "namespace", info.Namespace, "reason", info.Reason, "event reason", eventReason)
+			}
+		}
```

Comment:
This looks like the right logic, if the flag is not set, directly add the event to the queue, otherwise only add the event when the `reason` is also present in the `gen.emitEvents`